datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -627,20 +620,25 @@ pub fn update_hash(
     // calculate the hash values
     let hash_values = create_hashes(&keys_values, random_state, hashes_buffer)?;
 
+    if let Some(hash_map) = hash_map.as_any_mut().downcast_mut::<PruningJoinHashMap>() {
```

Comment:
I am open to ideas on this.

Reply:
It could be handled as well by a method on `JoinHashMapType` trait (void implementation for `JoinHashMap`)

Reply:
👍